### PR TITLE
add DiagnosticVariables.md

### PR DESCRIPTION
**Pull request by szy21**
# Description

- Add a separate file (DiagnosticVariables.md) for the list of diagnostic variables in CliMA.
- Point to DiagnosticVariables.md in the Diagnostics section in VariableList.md (not linked yet).

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by szy21** [comment]
@LenkaNovak Is there a list for diagnostic variables in the GCM? If so, maybe it's good to add it here.

**Comment by szy21** [comment]
@kpamnany I converted the diagnostic list to a table :)

File: docs/src/DiagnosticVariables.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+This document contains the diagnostic variables in CliMA.
+
+## LES Diagnostics
+
+### default
+
+| code name            | short name | description                                                                |
+|----------------------|------------|----------------------------------------------------------------------------|
+| ρ                    | ρ          | density                                                                    |
```
Comment (kpamnany) [inline docs/src/DiagnosticVariables.md:11]: Short name should be `rho`?

**Pull request merged**